homeassistant/components/icloud/account.py

```diff
@@ -165,6 +180,7 @@ def update_devices(self) -> None:
                 )
                 self._devices[device_id] = IcloudDevice(self, device, status)
                 self._devices[device_id].update(status)
+                new_device = True
```

Comment:
```python
                device = IcloudDevice(self, device, status)
                device.update(status)
                self._devices[device_id] = device
                device_registry = await self.hass.helpers.device_registry.async_get_registry()
                device_registry.async_get_or_create(
                    config_entry_id=self._entry.entry_id,
                    identifiers={(DOMAIN, device.unique_id)},
                    manufacturer="Apple",
                    model=device.device_model,
                    name=device.name,
                )
```

Should I do something like this instead ? (not for working because of function not async)